### ISSUE TITLE
Missing dependency: 'requests' - cannot provision instances on AWS

### ISSUE THREAD

**Issue by erlchamp** [saltstack/salt#51210]
I tried to deploy the instances on AWS but I'm getting errors/warnings. 

Failed to create instance using profile <function profile at 0x7f3137e4c2a8>, please check your configuration and also getting missing dependency, I am not sure if it is related.

### Description of Issue/Question

root@ip-172-31-18-244:~#salt-call state.apply test.infrastructure.aws.build_servers -l debug
```
root@ip-172-31-18-244:# salt-cloud --list-profiles pp1      
[WARNING ] Missing dependency: 'requests'. The requests driver requires 'ec2' to be installed.
[WARNING ] The cloud driver, 'ec2', configured under the 'pp1' cloud provider alias, could not be loaded. Please check your provider configuration files and ensure all required dependencies are installed for the 'ec2' driver.
In rare cases, this could indicate the 'ec2.get_configured_provider()' function could not be found.
Removing 'ec2' from the available providers list
```
```
root@ip-172-31-18-244:/srv/salt/test/infrastructure/aws# salt ip-172-31-18-244.ap-northeast-1.compute.internal pillar.items test.infrastructure.aws.build_servers
ip-172-31-18-244.ap-northeast-1.compute.internal:
    ----------
    test.infrastructure.aws.build_servers:
```

```
[ERROR   ] Failed to create instance test-pp1-k8s01 using profile <function profile at 0x7f3137e4c2a8>, please check your configuration
[INFO    ] Completed state [*-pp1-k8s01] at time 08:34:22.519527 (duration_in_ms=182.668)
[DEBUG   ] File /var/cache/salt/minion/accumulator/139849437138384 does not exist, no need to cleanup
[DEBUG   ] LazyLoaded state.check_result
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/minion', u'ip-172-31-18-244.ap-northeast-1.compute.internal', u'tcp://127.0.0.1:4506', u'aes')
[DEBUG   ] Initializing new AsyncAuth for (u'/etc/salt/pki/minion', u'ip-172-31-18-244.ap-northeast-1.compute.internal', u'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] LazyLoaded highstate.output
```

### Setup
```
/etc/salt/cloud.profiles.d# cat ec2-centos7-64bit.conf
#Asia Pacific (Seoul)
ec2-centos7-64bit-ap-northeast-2:
  provider: pp1
  image: ami-b6bb47d4

#Asia Pacific (Tokyo)
ec2-centos6-64bit-ap-northeast-1:
  provider: pp1
  image: ami-88b923e
```
```
/etc/salt/cloud.provider.d# cat ec2-centos7-64bit.conf
pp1:
  driver: ec2
  minion:
    master: 172.31.18.244

  ssh_interface: private_ips
    # CHANGE <ID> to AWS account ID
  id: 'A**********'
  # CHANGE <KEY> to AWS key
  key: '*********'

  private_key: /etc/salt/donw
  keyname: donw
  # CHANGE <LOCATION> to region such as eu-west-1
  location: ap-northeast-2
  # CHANGE <AZ> to availability_zone, normally the <LOCATION> and a letter, a to c. Default zone is a. example: eu-west-1a
  availability_zone: ap-northeast-2a
  del_root_vol_on_destroy: True
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
root@ip-172-31-18-244:/etc/salt/cloud.providers.d#  salt --versions-report
Salt Version:
           Salt: 2018.3.3

Dependency Versions:
           cffi: 1.11.5
       cherrypy: 3.5.0
       dateutil: 2.7.5
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 12 2018, 14:36:49)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1074-aws
         system: Linux
        version: Ubuntu 16.04 xenial

```

**Comment by dmurphy18** [comment]
@erlchamp Salt 2018.3.3 was update to rely on the python-requests provided by the RHEL 7 OS
python-requests.noarch 2.6.0-1.el7_1. It should be should be available from versions of RHEL 7 since v7.1

**Comment by erlchamp** [comment]
@dmurphy18 but I do have that package already installed

```
root@ip-172-31-18-244:/home/ubuntu# apt-get install python-requests
Reading package lists... Done
Building dependency tree
Reading state information... Done
python-requests is already the newest version (2.9.1-3ubuntu0.1)
```
